deployments/alphan/alphan.py

```diff
@@ -134,18 +168,30 @@ def __init__(self):
         self.logger.info('models loaded')
 
     @app.post("/predict", name="", summary="This endpoint detects nuclei positions in an image.")
-    async def predict(self, background_tasks: BackgroundTasks, input_image: UploadFile, model_name: str = 'unet_default', is_extract_regionprops: str = 'True',
-                      window: int=128):
-
-        # ref: https://stackoverflow.com/questions/63580229/how-to-save-uploadfile-in-fastapi
-        async with aiofiles.open(os.path.join(data_dir, input_image.filename), 'wb') as out_file:
-            while content := await input_image.read(1024):  # async read chunk
-                await out_file.write(content)  # async write chunk
+    async def predict(self, background_tasks: BackgroundTasks, input_image: UploadFile = None, model_name: str = 'unet_default', is_extract_regionprops: str = 'True',
+                      window: int=128, labcas_id: str = '', user: str=''):
+
+        if labcas_id!='':
+            # copy the file pointed by labcas_id to the data_dir, so we do not work directly on a file in labcas
+            shutil.copy(os.path.join(LabCAS_archive, labcas_id), os.path.join(data_dir, os.path.basename(labcas_id)))
+            resource = labcas_id
+            publish_to_labcas = True
+        else:
+            # ref: https://stackoverflow.com/questions/63580229/how-to-save-uploadfile-in-fastapi
+            async with aiofiles.open(os.path.join(data_dir, input_image.filename), 'wb') as out_file:
+                while content := await input_image.read(1024):  # async read chunk
```

Comment:
Great to see the walrus operator!

Reply:
@nutjob4life yups thanks to who wrote the actual code (see the reference) ;)